### PR TITLE
[Plugins] Read telemetry from the Deep Space Network

### PR DESCRIPTION
**Pull request by tobiasbrown**
Apologies for taking so long to complete this PR but what a pleasure it's been to work on it!  Thank you @akhenry for your assistance and to @evenstensberg and @naknode for your offers to help.  It'd be great to hear any feedback that you have.

This plugin adds a Deep Space Network folder to the object tree with folders for each site.

dsn-root

## Stations
Each location contains station information and folders for each dish.  Station telemetry can be viewed in a table or individually.

dsn-station

## Dishes
Each dish contains telemetry related to the:
* antenna
* signals it's sending and receiving
* targets it's communicating with

dsn-dish

### Antenna
Antenna telemetry can be viewed in a table or individually.

dsn-antenna

### Signals
Signal telemetry can only be viewed in a table.

dsn-signals

### Targets
Target telemetry can only be viewed in a table.

dsn-targets

### Known issues
* There may be some linting errors to fix.
* I've noticed tests in `DsnParserSpec.js` that expect empty results occasionally fail.  This may be a problem with my `beforeAll`and `afterAll` functions.
* Nothing indicates which objects in the tree are receiving data.  Are there features in the API to help with this?

### Author Checklist
1. Changes address original issue? Yes
2. Unit tests included and/or updated with changes? Yes
3. Command line build passes? Maybe.  See issues above.
4. Changes have been smoke-tested? Yes

### CODE REVIEW

File: src/plugins/dsn/plugin.js [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+                    parser = new DsnParser();
+
+                dsn = parser.parseXml(resp.request.responseXML);
+                config = dsn.data;
+            });
+    }
+
+    function getDsnData(domainObject) {
+        // Add the same query string parameter the DSN site sends with each request
+        var url = '/proxyUrl?url=' + encodeURIComponent(DSN_TELEMETRY_SOURCE + '?r=' + Math.floor(new Date().getTime() / 5000));
```
Comment (evenstensberg) [inline src/plugins/dsn/plugin.js]: I'd split this up to variables

File: src/plugins/dsn/plugin.js [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+                            if (lightTime < 60) {
+                                return lightTime.toFixed(2) + ' sec';
+                            } else if (lightTime < 3600) {
+                                return (lightTime / 60).toFixed(2) + ' minutes';
+                            } else if (lightTime < 86400) {
+                                return (lightTime / 3600).toFixed(2) + ' hours';
+                            } else if (lightTime < 604800) {
+                                return (lightTime / 86400).toFixed(2) + ' days';
+                            } else {
+                                return (lightTime / 604800).toFixed(2) + ' weeks';
```
Comment (evenstensberg) [inline src/plugins/dsn/plugin.js]: Maybe extract all these ints to `CONST SOME_VAL`

File: src/plugins/dsn/res/http.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
+(function (root, factory) {
```
Comment (evenstensberg) [inline src/plugins/dsn/res/http.js]: Don't we have a built-in http lib?

File: src/plugins/dsn/plugin.js (hunk 2 above)
Comment (naknode) [reply src/plugins/dsn/plugin.js]: @evenstensberg Don't think we're doing ES6 with all of the `var` going on...

Comment (evenstensberg) [reply src/plugins/dsn/plugin.js]: sorry, so what I'm meaning is: `var CONST_GOD_SPEED = 123`

**Comment by akhenry** [comment]
@tobiasbrown This looks really awesome, thanks for all your effort here! We will review the PR as soon as we're able and get you some feedback.

**Comment by jvigliotta** [comment]
@tobiasbrown Hello! As you may have noticed, this fell off our radar. We're going to include a reviewing this as a task in our next sprint. We'll get back to you soon!

Thanks, this looks awesome!

**Comment by jvigliotta** [comment]
After some conversation with the team, we think this would be better off as a plugin you own in it's own repository. We love it though and we'll definitely link to it in our [plugins](https://nasa.github.io/openmct/plugins/) page on: https://nasa.github.io/openmct/ that being said, we'll still review this. We'll keep you updated. Thanks again!

Also, once it's complete, you should most definitely post it in the 
 [Show and Tell](https://github.com/nasa/openmct/discussions/categories/show-and-tell) category of our Discussions section.

**Review by akhenry — CHANGES_REQUESTED** [review]
@tobiasbrown This is a really awesome submission. Code is super clean, and has great test coverage. Nice work. I've left a couple of suggestions where the Open MCT platform could do some of the heavy lifting that is currently being done in plugin code, and also some ways in which it could be organized differently that might help with long term maintainability.

Just a general note, we recommend using ES6 these days, even though a lot of our legacy code is pre-ES6.

This is a really nice submission though, great work. As @jvigliotta has said, this would be better maintained as a separate repository, rather than a core Open MCT plugin. That way you will not have to wait for us to approve any changes to the plugin. It also helps us to keep core Open MCT as lean as possible, because right now we don't have the ability to do configurable custom builds.

Let us know when you have created a separate repository for this and we will link to it from [our plugins page](https://nasa.github.io/openmct/plugins/).

Again, great work here! Sorry it took us so long to review this.

**Comment by tobiasbrown** [comment]
Thanks for the feedback @jvigliotta and taking the time to review @akhenry .  I'll let you know once I've moved this to a separate repository.

**Comment by jvigliotta** [comment]
@tobiasbrown awesome! Thanks again for this work and don't forget to let us know when it's complete and we'll add it to our plugins page for the official Open MCT site. I'll close this for now.

**Pull request closed without merging**